streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -133,112 +129,179 @@
 
 
 
+    public static final String SECURITY_PROTOCOL_CONFIG = CommonClientConfigs.SECURITY_PROTOCOL_CONFIG;
```

Comment:
Just had another discussion with @hjafarpour offline. And after that I am now lining towards keeping these configs on the `StreamsConfig` layer as well for the following two reasons:

1. All these `CommonClientConfigs` are defined both in producer and consumer configs as well, but their config definitions may have different default values. It is not clear which default values we should use then if users do not specify customized values.

2. Right now we have the `prefix` for producer and consumer to differentiate those common configs. So following this pattern if we define the same config `X` also in StreamsConfig, we can then allow users to specify different values for `X` for streams' admin client, and producer / consumer client separately, as:

```
producer.X: 100
consumer.X: 200
X: 300
```

And without the prefixed overriding:

```
X: 300
```

Then all three clients will use the same value, which is reasonable.